src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/CachedExpressionCompiler.cs

```diff
@@ -38,117 +38,109 @@ private static class Compiler<TModel, TResult>
 
             public static Func<TModel, TResult> Compile(Expression<Func<TModel, TResult>> expression)
             {
-                if (expression == null)
+                switch (expression.Body)
```

Comment:
`Debug.Assert(expression != null)`

Reply:
Looks like this is almost ready to go. Just address these two older comments&hellip;